controller/recurring_job_controller.go

```diff
@@ -318,6 +369,10 @@ func (control *RecurringJobController) newCronJob(recurringJob *longhorn.Recurri
 	}
 	registrySecret := registrySecretSetting.Value
 
+	// Cronjob sends SIGTERM to the container when RecurringJob gets removed.
+	// Need to allow enough time for termination for recurring job auto-attached
+	// volume to finish and detach volume.
+	terminationGracePeriodSeconds := int64(types.DefaultRecurringJobTerminationGracePeriodSeconds)
```

Comment:
Whats the issue for adding this?

Reply:
Ok checked again the commit history. Why not fix at e2e to wait until the volume detached but use lengthen the grace period here instead ?

Reply:
When the RecurringJob gets removed, its CronJob will send SIGTERM to the recurring job pod. The pod could get force terminated when it doesn't finish within the grace period, and this is likely to result:
1. Incomplete backups.
2. Volume not detached when it is set to allow recurring-job while volume id detached.
So I feel this is a better solution than fixing the e2e.

Reply:
What u want to achieve is to try detach the volume before the pod gets terminated. 

As discussed, think about if trying to detach the volume when receiving the termination signal instead of having a long termination period to keep the pod running until the backup finished and the volume detached, even the corresponding recurring job has been deleted. Because this will probably cause some confusion when there is a new recurring job being created for that volume, but it will never be executed successfully because a terminating job still running.

Also, a failed backup is reasonable if the recurring job is deleted, so probably we shouldn't do anything to make it work in this case.

However know your intention.

Reply:
The ultimate solution for this issue is volume-controller refactoring. After the refactoring, there will be a separate controller handling all kinds of attachment & detachment.

Reply:
> P.S. in any solution, it's still possible to fail to detach the volume at the end when passing the termination period. It's fine but need to note.

I've rethought about catching the termination signal. IMO it might not be a good solution because it still means the volumes needs to be detached within the 30 seconds grace time. It is possible for a large number of volumes the detach could exceed the grace period. So I've think moving the logic to the recurring job controller might be a better solution.